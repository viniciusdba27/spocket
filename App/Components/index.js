import SearchBox from './SearchBox'
import Card from './Card'
import Loader from './Loader'

export { SearchBox, Card, Loader }
