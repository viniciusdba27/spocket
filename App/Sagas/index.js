import { takeLatest, all } from 'redux-saga/effects'
import { ExampleTypes } from 'App/Stores/Example/Actions'
import { StartupTypes } from 'App/Stores/Startup/Actions'
import { fetchResults } from './ExampleSaga'
import { startup } from './StartupSaga'
import { getAppWindowSize } from './AppLayoutSaga'

import { AppLayoutTypes } from 'App/Stores/AppLayout/Actions'

export default function* root() {
  yield all([
    takeLatest(StartupTypes.STARTUP, startup),

    takeLatest(AppLayoutTypes.GET_WINDOW_SIZE, getAppWindowSize),
    takeLatest(ExampleTypes.FETCH_RESULTS, fetchResults),
  ])
}
