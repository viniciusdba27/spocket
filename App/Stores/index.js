import { combineReducers } from 'redux'
import configureStore from './CreateStore'
import rootSaga from 'App/Sagas'
import { reducer as ExampleReducer } from './Example/Reducers'
import { reducer as AppLayout } from './AppLayout/Reducers'

export default () => {
  const rootReducer = combineReducers({
    example: ExampleReducer,
    appLayout: AppLayout,
  })

  return configureStore(rootReducer, rootSaga)
}
