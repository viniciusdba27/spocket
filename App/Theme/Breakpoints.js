const Layouts = {
  S: 'S',
  M: 'M',
  L: 'L',
  XL: 'XL',
}

const breakpoints = [500, 900, 1140]

const getLayout = function(width) {
  if (width > breakpoints[2]) {
    return Layouts.XL
  }
  if (width > breakpoints[1] && width <= breakpoints[2]) {
    return Layouts.L
  }
  if (width > breakpoints[0] && width <= breakpoints[1]) {
    return Layouts.M
  }
  return Layouts.S
}

export default {
  Layouts,
  getLayout,
}
