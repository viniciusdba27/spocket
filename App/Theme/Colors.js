/**
 * This file contains the application's colors.
 *
 * Define color here instead of duplicating them throughout the components.
 * That allows to change them more easily later on.
 */

export default {
  transparent: 'rgba(0,0,0,0)',
  // Example colors:
  white: '#ffffff',
  text: '#212529',
  primary: '#8144e5',
  success: '#28a745',
  error: '#dc3545',

  gray1: '#e1e1e1',
  gray2: '#7e768f',
}
