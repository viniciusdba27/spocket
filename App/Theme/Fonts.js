import { StyleSheet } from 'react-native'

const size = {
  h1: 38,
  h2: 34,
  h3: 30,
  input: 18,
  regular: 16,
  medium: 12,
  small: 11,
}

export default StyleSheet.create({
  bold: {
    fontWeight: 'bold',
  },
  h1: {
    fontSize: size.h1,
  },
  h2: {
    fontSize: size.h2,
  },
  h3: {
    fontSize: size.h3,
  },
  medium: {
    fontSize: size.medium,
  },
  normal: {
    fontSize: size.regular,
  },
  small: {
    fontSize: size.small,
  },
})
