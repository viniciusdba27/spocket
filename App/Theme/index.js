import Colors from './Colors'
import Fonts from './Fonts'
import Metrics from './Metrics'
import Images from './Images'
import Helpers from './Helpers'
import ApplicationStyles from './ApplicationStyles'
import Breakpoints from './Breakpoints'

export { Colors, Fonts, Images, Metrics, Helpers, ApplicationStyles, Breakpoints }
