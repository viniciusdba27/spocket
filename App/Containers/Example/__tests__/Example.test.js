/**
 * @format
 */

import 'react-native'
import React from 'react'
import ExampleScreen from '../index'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Breakpoints } from 'App/Theme'

const mockStore = configureMockStore()
const store = mockStore({
  example: {
    resultsIsLoading: false,
    results: [],
  },
  appLayout: {
    breakpoint: Breakpoints.S,
  },
})

const results = jest.useFakeTimers()

// TODO: test page with results

it('renders correctly', async () => {
  const component = renderer
    .create(
      <Provider store={store}>
        <ExampleScreen />
      </Provider>
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})
