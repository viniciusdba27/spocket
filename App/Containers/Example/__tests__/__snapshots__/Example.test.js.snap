// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "marginHorizontal": 15,
      },
      Object {
        "marginVertical": 15,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 42.5,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Product Search
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 15,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "paddingLeft": 0,
                  "paddingRight": 75,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#e1e1e1",
                    "borderWidth": 2,
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  editable={true}
                  onChangeText={[Function]}
                  placeholder="Enter keywords..."
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "height": 0,
                      "minHeight": 40,
                      "paddingHorizontal": 15,
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "height": 0,
                    "margin": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "height": "100%",
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 75,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "center",
                      "padding": 8,
                      "paddingVertical": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontSize": 15,
                        },
                        undefined,
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï€‚
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
